templates/config.m4.template

```diff
@@ -38,13 +38,14 @@
     PHP_SUBST(GRPC_SHARED_LIBADD)
   <%
     srcs = []
-    for src in php_config_m4.src:
-      srcs.append(src)
-    for lib in libs:
-      if lib.name in php_config_m4.get('deps', []) and lib.name != 'z':
-        for src in lib.src:
-          srcs.append(src)
-    srcs = sorted(srcs)
+    srcs.extend(php_config_m4.src)
+    php_deps = php_config_m4.get('deps', [])
+    lib_maps = {lib.name: lib for lib in libs}
+    for dep in php_deps[:]:
+      php_deps.extend(lib_maps[dep].transitive_deps)
+    for dep in list(set(php_deps) - set(('z',))):
```

Comment:
Because set with the string could be misleading, I prefer to use `{'z'} here.
```
>>> set('abc')
set(['a', 'c', 'b'])
>>> set({'abc'})
set(['abc'])
```